### PR TITLE
AZP: Retain env for manual repro

### PR DESCRIPTION
**Pull request by Alexey-Rivkin**
## What
Retain working dirs for the three most recent builds for manual repro attempts.
Depends on https://github.com/openucx/ucx/pull/9431

### CODE REVIEW

**Review by ivankochin — APPROVED** [review]

File: buildlib/azure-pipelines-perf.yml [hunk 1]
```diff
@@ -128,5 +130,11 @@ stages:
             clean: true
           - bash: |
               set -x
+              echo 'Retain 3 latest workdirs for manual repro; clean the rest'
+              wrk_dirs="/hpc/scrap/azure/$(Build.DefinitionName)/"
+              for dir in $(ls -lt "$wrk_dirs" | tail -n +4 | awk '{print $NF}'); do
```
Comment (artemry-nv) [inline buildlib/azure-pipelines-perf.yml:135]: Potentially unsafe - please check shellcheck warnings

File: buildlib/azure-pipelines-perf.yml [hunk 2]
```diff
@@ -128,5 +130,11 @@ stages:
...
+                  echo "Removing old workdir: $wrk_dirs$dir"
+                  rm -rf "$wrk_dirs$dir"
```
Comment (artemry-nv) [inline buildlib/azure-pipelines-perf.yml:137]: Consider changing to more readable version:
`rm -rf "${wrk_dirs}/${dir}"`

**Review by artemry-nv — APPROVED** [review]

File: buildlib/azure-pipelines-perf.yml (hunk 1 above)
Comment (Alexey-Rivkin) [reply buildlib/azure-pipelines-perf.yml:135]: We can safely disregard shellcheck[SC2012](https://www.shellcheck.net/wiki/SC2012), as we do not generate non-alphanumeric filenames.

File: buildlib/azure-pipelines-perf.yml (hunk 2 above)
Comment (Alexey-Rivkin) [reply buildlib/azure-pipelines-perf.yml:137]: The syntax suggested raises shellcheck [SC2115](https://www.shellcheck.net/wiki/SC2115)

Comment (Alexey-Rivkin) [reply buildlib/azure-pipelines-perf.yml]: Some extra curly brackets never hurt though, added.

**Comment by Alexey-Rivkin** [comment]
@yosefe could you please review?

File: buildlib/azure-pipelines-perf.yml [hunk 3]
```diff
@@ -128,5 +130,13 @@ stages:
...
           - bash: |
               set -x
+              echo 'Retain 3 latest workdirs for manual repro; clean the rest'
+              wrk_dirs="/hpc/scrap/azure/$(Build.DefinitionName)/"
+              for dir in $(ls -lt "$wrk_dirs" | tail -n +4 | awk '{print $NF}'); do
+                  echo "Removing old workdir: ${wrk_dirs}${dir}"
+                  rm -rf "${wrk_dirs}${dir}"
+              done
+
+
```
Comment (yosefe) [inline buildlib/azure-pipelines-perf.yml:140]: should we remove line 141?

Comment (Alexey-Rivkin) [reply buildlib/azure-pipelines-perf.yml:140]: OMG, it literally defeats the PR's purpose!
Thanks a lot for caching it.

**Review by yosefe — APPROVED** [review]

**Comment by yosefe** [comment]
@Alexey-Rivkin pls squash

**Pull request merged**